python-app/docker/action.yml

```diff
@@ -0,0 +1,61 @@
+name: Build
+description: Build and publish image
+
+runs:
+  using: "composite"
+
+  steps:
+    - name: Check if step is needed
+      run: |
+        echo docker_needed=${{ startsWith(github.ref, 'refs/tags/') || startsWith(github.ref, 'refs/heads/feature/') || github.ref == 'main' }} >> $GITHUB_ENV
```

Comment:
It made me think of:

This kind of thing can be done by writing to ENV or using step outputs. The outcome is very similar IMO, and I don't have a good rule of thumb on this. But as you work on more GAs, let me know if you feel like one or the other is better in your opinion.